### DISCUSSION TITLE
how to improve the rate of compress ?

### DISCUSSION THREAD

**Discussion by Tovi163** [klauspost/compress#712] [Q&A] [answered]
hi  all 

I  see  many parameters of zstd options like 
<img width="515" alt="image" src="https://user-images.githubusercontent.com/15190164/207277816-de046761-6522-4e6e-92b2-4aecda50ea26.png">

how to update these parameters (except the level) for improving the rate of zstd compress

thanks!

**Comment by klauspost** [comment, marked as answer]
This is not exported. Use [zstd.With....](https://pkg.go.dev/github.com/klauspost/compress@v1.15.13/zstd#EOption) options to change the encoder options.

**Reply by Tovi163** [reply]
> This is not exported. Use [zstd.With....](https://pkg.go.dev/github.com/klauspost/compress@v1.15.13/zstd#EOption) options to change the encoder options.

thanks for your answer. I know that use zstd.With.... can change the options.
But what i want to know is, if there are some suggestions for these parameters(except the level), such as enlarge the value of blockSize can improve the rate of zstd

thanks a lot

**Reply by klauspost** [reply]
The two main tweaks are `WithEncoderLevel()`, which will tweak window size as well. 

Block size is already at maximum, unless you use `SpeedFastest`. You can play with Window Size as well, which is 4MB, 8MB, 16MB and 32MB for the 4 compression levels by default.

**Reply by Tovi163** [reply]
Got it, thank a lot
